README.md

```diff
@@ -1,37 +1,63 @@
 # Eclipse Che on OpenShift
 
+## Table Of Content
+
+* [How to build the RedHat Che distribution](#how-to-build-the-redhat-che-distribution)
+* [How to build the upstream openshift-connector branch for development purposes](#how-to-build-the-upstream-openshift-connector-branch-for-development-purposes)
+* [How to run Che on OpenShift](#how-to-run-che-on-openshift)
+
 ## How to build the RedHat Che distribution
 
+### TL;DR
+
+There is a good chance that you want to build rh-che with the dashboard but without keycloak support (because it's easier to test) and as fast as possible :bowtie::
+
+```
+git clone https://github.com/redhat-developer/rh-che
+cd rh-che
+mvn -DwithoutKeycloak `# disable Keycloak support` \
+    -Pfast            `# skip tests and other verifications` \
+    clean install
+```
+
+If you have already cloned [eclipse/che](https://github.com/eclipse/che) and built [openshift-connector branch](https://github.com/eclipse/che/tree/openshift-connector) you will want to:
+
+```
+UPSTREAM_CHE_PATH=/path/to/upstream/che
+mvn -DwithoutKeycloak                          `# disable Keycloak support` \
+    -Pfast                                     `# skip tests and other verifications` \
+    -DlocalCheRepository=${UPSTREAM_CHE_PATH}  `# get already built upstream Che` \
+    clean install
+```
+
 #### What is the RedHat Che distribution
 
-The RedHat distribution of Eclipse Che is a RedHat-specific packaging of the Che assemblies
-that allows adding some RedHat specific plugins / behaviors upton the standard upstream Che
+The RedHat distribution of Eclipse Che is a RedHat-specific packaging of Che assemblies
+that adds some RedHat specific plugins / behaviors upton the standard upstream Che
 distribution. It is currently based on the `openshift-connector` branch of the upstream
-Che repository.
+Che repository. The RedHat distribution powers [openshift.io](https://openshift.io) developers workspaces.
 
 RedHat modifications against the upstream Che include:
-- the ability to disable the Dashboard (and remove the *Go to Dashboard* button from the Che IDE)
-- The Keycloak integration
-- The Bayesian integration (soon)
+- The ability to disable the Dashboard (and remove the *Go to Dashboard* button from the Che IDE)
+- Keycloak integration
+- [fabric8-analytics Language Server](https://github.com/fabric8-analytics/fabric8-analytics-lsp-server) 
 
 #### How the RedHat Che distribution build works
 
-The RedHat Che distribution build is a *maven* build. By default, it *automatically* takes care 
```

Comment:
At the beginning, the initial test was somewhat similar to the one you propose. Then I tried to make it clearer and more explicit due to a comment:
https://github.com/redhat-developer/rh-che/commit/bbeb548236003ce0904133f1e039bb6a4b382d4e
It seemed it was needed to make even clearer clear that the checkout of the upstream project is *automatic*, and that this is *only by default*.
   
Don't know if this insistance is still relevant.

Anyway for me it's OK in boith cases, since we tell it just after (*However ...*)